GliaWidgets/ViewController/Survey/Components/Survey.ButtonView.swift

```diff
@@ -0,0 +1,75 @@
+import UIKit
+
+extension Survey {
+    final class ButtonView: View {
+
+        enum State {
+            case active, highlighted, selected
+        }
+
+        struct Props {
+            let title: String
+            let state: State
+            let onSelection: () -> Void
+        }
+
+        let value = UILabel().makeView {
+            $0.clipsToBounds = true
+            $0.textAlignment = .center
+            $0.layer.cornerRadius = 4
+            $0.layer.borderWidth = 1
+            $0.layer.borderColor = UIColor.gray.cgColor
+        }
+
+        var props: Props { didSet { render() } }
+
+        init(props: Props = .init(title: "", state: .active, onSelection: {})) {
+            self.props = props
+            super.init()
+        }
+
+        override func setup() {
+            super.setup()
+
+            addSubview(value)
+            let gesture = UITapGestureRecognizer(target: self, action: #selector(tapGestureAction))
+            addGestureRecognizer(gesture)
+
+            render()
+        }
+
+        override func defineLayout() {
+            super.defineLayout()
+
+            var constraints = [NSLayoutConstraint](); defer { constraints.activate() }
+            constraints += value.topAnchor.constraint(equalTo: topAnchor, constant: 1)
+            constraints += value.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 1)
+            constraints += value.trailingAnchor.constraint(equalTo: trailingAnchor, constant: 1)
+            constraints += value.bottomAnchor.constraint(equalTo: bottomAnchor, constant: 1)
+        }
+
+        func render() {
+            value.text = props.title
+            switch props.state {
+            case .active:
+                value.layer.borderColor = UIColor.blue.cgColor
```

Comment:
Colors are to be provided for themes, do I get it right?

Reply:
yes, I'll move out all of them into survey theme.